modules/benchmark/R/clone_run.R

```diff
@@ -0,0 +1,115 @@
+##-----------------------------------------------------------------------------#
+##' Produce a "cloned" run from the settings of a completed run
+##' 
+##' @name clone_run 
+##' @title Clone Run
+##' @param settings settings or multisettings object 
+##' @param model model id 
+##' @param con database connection
+##' @export 
+##' 
+##' @author Betsy Cowdery 
+##' @importFrom dplyr tbl filter rename collect select 
+
+clone_run <- function(settings, model, con){
+  
+  # For testing: 
+  # bety <- betyConnect("web/config.php")
+  # con <- bety$con
+  # settings <- read.settings("/fs/data2/output/PEcAn_1000008371/pecan.CHECKED.xml")
+  # model = 1000000020
+  
+  settings.old <- settings 
+  settings.new <- list()
+  
+  # Copy over settings that won't change
+  settings.new$meta.analysis <- settings.old$meta.analysis
+  settings.new$ensemble <- settings.old$ensemble
+  settings.new$run$site <- settings.old$run$site
+  settings.new$run$start.date <- settings.old$run$start.date
+  settings.new$run$end.date <- settings.old$run$end.date
+  
+  # New model info
+  model.new <- tbl(bety, "models") %>% filter(model == id) %>% collect()
+  # List of all/required inputs for the new model 
+  inputs_all <- tbl(bety, "modeltypes_formats") %>% filter(model.new$modeltype_id == modeltype_id) %>% select(one_of("tag", "required")) %>% collect
+  inputs_req <- inputs_all$tag[inputs_all$required]
+  
+  #------ Update model ------#
+  
+  settings.new$model$id <- model.new$id
+  
+  #------ Update inputs ------#
+  
+  for(tag in names(settings.old$run$inputs)){
+    
+    # Check that the given input is required for the new model
```

Comment:
I think the logic below is right, but the comment here suggests you're only matching 'required' tags, rather than converting all tags you can regardless of whether they're optional or required